roles/validations/tasks/edpm/bmh_compute_replace.yml

```
@@ -0,0 +1,196 @@
+# This job tests the functionality of the openstack-operator to replace a
+# bmh compute node.
+#
+# This job was created to satisfy:
+# https://issues.redhat.com/browse/OSPRH-15061
+
+- name: Get name of nodeset containing bmh node to be used as spare node
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} get bmh {{ cifmw_validations_bmh_spare_leaf_label }} --no-headers | awk '{print $3}'
+  register: bmh_nodeset_name
+
+- name: Verify bmh node being used as spare is provisioned before scale down
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} get bmh {{ cifmw_validations_bmh_spare_leaf_label }}
+  register: bmh_nodes_before_scale_down
+  failed_when: '"provisioned" not in bmh_nodes_before_scale_down.stdout'
```

```suggestion
      oc -n {{ cifmw_validations_namespace }} get bmh {{ cifmw_validations_bmh_spare_leaf_label }} -o jsonpath='{.status.provisioning.state}'
  register: bmh_nodes_before_scale_down
  failed_when: bmh_nodes_before_scale_down != "provisioned"
```

I think that just makes the `failed_when` conditional a little more reliable. Just in case the stdout is something unexpected that just happens to container `provisioned`.

---

I'll try it. From command line it looks like it should work:
oc -n openstack get bmh leaf0-1 -o jsonpath='{.status.provisioning.state}'
provisioned

---

done